cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/RemoteArtifactRepositoryReader.java

```diff
@@ -93,6 +100,36 @@ public ArtifactDetail getArtifact(Id.Artifact artifactId) throws Exception {
     return artifactDetail;
   }
 
+  @Override
+  public InputStream newInputStream(Id.Artifact artifactId) throws IOException, NotFoundException {
+    String namespaceId = artifactId.getNamespace().getId();
+    ArtifactScope scope = ArtifactScope.USER;
+    // Cant use 'system' as the namespace in the request because that generates an error, the namespace doesnt matter
+    // as long as it exists. Using default because it will always be there
+    if (ArtifactScope.SYSTEM.toString().equalsIgnoreCase(namespaceId)) {
+      namespaceId = "default";
+      scope = ArtifactScope.SYSTEM;
+    }
+    String url = String.format("namespaces/%s/artifacts/%s/versions/%s/download?scope=%s",
+                               namespaceId,
+                               artifactId.getName(),
+                               artifactId.getVersion(),
+                               scope);
+
+    HttpURLConnection urlConn = remoteClient.openConnection(HttpMethod.POST, url);
+    urlConn.setChunkedStreamingMode(64 * 1024);
```

Comment:
This is not needed. This is for streaming large request body, not for reading in response body. By default reading of response body is always streaming through the InputStream.

Reply:
I see, removed it.